### PR TITLE
Asman Umbetov Tasks 91-100

### PR DESCRIPTION
**Pull request by fastndead**

### CODE REVIEW

File: tasks/91_make-a-person/make-a-person.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * @description
```
Comment (akimovaa) [inline tasks/91_make-a-person/make-a-person.js:2]: Work with private fields - smth like a   
<code>constructor(fullName) {
    [this._firstName, this._lastName] = fullName.split(' ');
  }
</code>

File: tasks/96_greatest-common-divisor/greatest-common-divisor.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * @description
+ * Calculates greatest common divisor of two numbers
+ * @example
+ * // returns 9
+ * greatestCommonDivisor(36, 45)
+ * @param {Number} a
+ * @param {Number} b
+ * @return {Number}
+ */
+export function greatestCommonDivisor(a, b) {
```
Comment (akimovaa) [inline tasks/96_greatest-common-divisor/greatest-common-divisor.js:11]: redundant code. try to find more logical and simple solution.

Comment (akimovaa) [reply tasks/96_greatest-common-divisor/greatest-common-divisor.js:11]: Recursion

File: tasks/97_least-common-multiple/least-common-multiple.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * @description
+ * Searches for greatest common divisor
+ * @example
+ * // returns 6
+ * gcd(54, 24)
+ * @param {Number} a
+ * @param {Number} b
+ * @return {Number}
+ */
+function gcd(a, b) {
```
Comment (akimovaa) [inline tasks/97_least-common-multiple/least-common-multiple.js]: image

**Review by akimovaa — DISMISSED** [review]

File: tasks/98_linked-list/linked-list.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  /**
+   * removes given value from list
+   * @param {Object} valueToRemove
+   * @return {boolean} - if value was found and removed
+   */
+  removeItem(valueToRemove) {
+    let foundValue = this.find(valueToRemove);
+    if (foundValue !== null) {
```
Comment (vovcyan) [inline tasks/98_linked-list/linked-list.js]: Rewrite it

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/91_make-a-person/make-a-person.spec.js [hunk 5]
```diff
@@ -36,10 +36,6 @@ xdescribe('Make a person', () => {
         expect(bob.getFullName()).toBe('Haskell Ross');
     });
 
-    test('bob.getFullName() should return "Haskell Ross" after bob.setLastName("Curry")', () => {
```
Comment (vovcyan) [inline tasks/91_make-a-person/make-a-person.spec.js:39]: For what?

File: tasks/99_stack/stack.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    this.size = 0;
+  }
+
+  /**
+   * pushes element in stack
+   * @param {Object} value
+   */
+  push(value) {
+    const newNode = new Node(value);
+    if (!this.last ) {
```
Comment (vovcyan) [inline tasks/99_stack/stack.js]: What the space it?

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/98_linked-list/linked-list.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
+/**
+ * Represents linked list data structure
+ */
+export class LinkedList {
+  /**
+   * creates new empty linked list
+   */
+  constructor() {
+    this.head = null;
+    this.tail = null;
```
Comment (alexkond) [inline tasks/98_linked-list/linked-list.js:10]: Please, read about linked lists. It doesn't contain a link to the tail. Only head.

File: tasks/98_linked-list/linked-list.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+  /**
+   * creates new node
+   * @param {Object} value
+   * @param {Object} next
+   * @param {Object} prev
+   */
+  constructor(value, next = null, prev = null) {
+    this.value = value;
+    this.next = next;
+    this.prev = prev;
```
Comment (alexkond) [inline tasks/98_linked-list/linked-list.js]: Linked list nodes should store link only to the next node, not to previous.

**Review by vovcyan — DISMISSED** [review]

**Review by alexkond — CHANGES_REQUESTED** [review]

File: tasks/98_linked-list/linked-list.js (hunk 7 above)
Comment (fastndead) [reply tasks/98_linked-list/linked-list.js:10]: It wasn't my idea, tests do actually require tail to be an actual property of the linked list

File: tasks/98_linked-list/linked-list.js (hunk 8 above)
Comment (fastndead) [reply tasks/98_linked-list/linked-list.js]: Got it, Fixed it. I guess my linked list was doubly linked

**Review by alexkond — APPROVED** [review]

**Review by savchukm — APPROVED** [review]

**Pull request merged**